PR-4559 Update S3 docs to include multipart_chunksize guidance (#729)
asfadmin/thin-egress-app: docs/s3access.md

[Using temporary credentials with AWS resources](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_use-resources.html)
for more information on how to use your temporary credentials.

### Multipart Chunksize

> The AWS multipart_chunksize by default is 8MB, it is recommended that a multipart_chucksize of 128MB-1GB is used. This is included in the example below. Addtional resources can be found from [AWS](https://docs.aws.amazon.com/cli/latest/topic/s3-config.html#configuration-values)

Reviewer: Is it MB or MiB?

Author: the docs are showing examples with MB. so I should swap my top to be 1000?. 1MB = 1000 Kb and 1MiB = 1024 Kib.

> The AWS `multipart_chunksize` by default is 8MB, it is recommended that a `multipart_chuncksize` of 128MB-1GB is used. This is included in the example below. Additional resources can be found from [AWS](https://docs.aws.amazon.com/cli/latest/topic/s3-config.html#configuration-values)

Reviewer: Still misspelled
```suggestion
The AWS `multipart_chunksize` by default is 8MB, it is recommended that a `multipart_chunksize` of 128MB-1GB is used. This is included in the example below. Additional resources can be found from [AWS](https://docs.aws.amazon.com/cli/latest/topic/s3-config.html#configuration-values)
```

> The AWS `multipart_chunksize` by default is 8MB, it is recommended that a `multipart_chuncksize` of 128MB-1GB is used. This is included in the example below. Additional resources can be found from [AWS](https://docs.aws.amazon.com/cli/latest/topic/s3-config.html#configuration-values)

Reviewer: ```suggestion
The AWS `multipart_chunksize` by default is 8MB, it is recommended that a `multipart_chuncksize` of 128MB-1GB is used. This is included in the example below. Additional resources can be found from [AWS](https://docs.aws.amazon.com/cli/latest/topic/s3-config.html#configuration-values).
```

> The AWS `multipart_chunksize` by default is 8MB, it is recommended that a `multipart_chuncksize` of 128MB-1GB is used. This is included in the example below. Additional resources can be found from [AWS](https://docs.aws.amazon.com/cli/latest/topic/s3-config.html#configuration-values)

Reviewer: I find some of the phrasing/punctuation to be a bit awkward. Maybe something like this would be better. I also wonder if we should be explaining why the bigger limit is better?
```suggestion
The AWS `multipart_chunksize` is 8MB by default. It is recommended to increase this to between 128MB-1GB as shown in the example below. Additional resources can be found in the [AWS documentation](https://docs.aws.amazon.com/cli/latest/topic/s3-config.html#configuration-values)
```

import json
import urllib.request

> MB = 1024 ** 2

Reviewer: Pretty sure
```suggestion
MiB = 1024 ** 2
```

Author: there is not direct guidance from AWS on what the unit is. I think in this case I did use a power of 2 for the 1024 so it should be MiB.

Author: I updated it to be 1000 and use MB as the docs from AWS do.

    # Lambda needs to have permission to upload to destination bucket
    upload_client = boto3.client("s3")

    # Set up transfer config object
    transfer_config = {

>         multipart_chunksize= 128 * MB

Reviewer: PEP 8
```suggestion
        multipart_chunksize = 128 * MB
```
